src/message.rs

```diff
@@ -760,6 +763,11 @@ impl Message {
             .await
             .is_ok()
     }
+
+    /// Gets the error of the message.
```

Comment:
I genuinely do not know *what* and "error of the message" is and don't even have a decent guess at it.  When does a message have an error, what does this mean?  When can I expect a message to have an error and when not?  Can I set errors myself or do they only come from within DC?

I think for a public API explaining the concepts behind them is valuable as well.  A lot of the time the deltachat.h file tries to do this, but the doc comment there also doesn't really explain this.  Could you describe more about the concept and explain these things?  Or if this is already written clearly somewhere at least refer to it?

Reply:
It was added by @Hocuri, error is added when you get NDN.

Reply:
I just made another change which rustifies the code a bit more and also makes it more clear what kind of errors we can expect by introducing an enum `MessageError` with both possible values

Reply:
> For received messages it could be that it failed to decrypt or when it did decrypt that the signature failed. I'm a little surprised this is stored in the DB, as it seems that this could change as this mostly depends on keys which may change over time

Whe don't try to decrypt already-received messages when we import a new private key. (except if I got something completely wrong). So, no, this does not change.

> For sent messages it is a message extracted from the non-delivery-notification if there was one for a given message?

Yes.